PBM-1363 Release notes 2.6.0 (#198)
percona/pbm-docs: docs/release-notes/2.6.0.md

You can now redefine node priorities not only for making backups but also for point-in-time recovery oplog slices. For example, you can instruct PBM to save oplog slices either from a dedicated node or from nodes in the geographically closest location, thus reducing network latency. 

See the [Adjust node priority for oplog slices](../usage/pitr-tutorial.md#adjust-node-priority-for-oplog-slices) section for guidelines.

### Additional control over PBM command execution

You can now configure the waiting time for a command execution by passing the `--wait-time` flag for the following commands:

* pbm config --force-resync --wait --wait-time="5m"
* pbm backup --wait --wait-time="5m"
* pbm restore --wait --wait-time="5m"
* pbm oplog-replay --wait --wait-time="5m"
* pbm profile add --sync --wait --wait-time="5m"
* pbm profile remove --wait --wait-time="5m"
* pbm profile sync --wait --wait-time="5m"
* pbm delete-pitr --older-than --wait --wait-time="5m"
* pbm cleanup --older-than --wait --wait-time="5m"

This way you have more control over the PBM operation. This enhancement also improves the error logging.

### Snapshot-based physical backups are generally available

> With snapshot-based physical backups now generally available, you can use this functionality in production environments and enjoy all the benefits of faster restores from snapshots with almost immediate access to data. 

Reviewer: Please add the link to learn more about snapshot-based physical backup ;)

Author: Done

* pbm profile remove --wait --wait-time="5m"
* pbm profile sync --wait --wait-time="5m"
* pbm delete-pitr --older-than --wait --wait-time="5m"
* pbm cleanup --older-than --wait --wait-time="5m"

This way you have more control over the PBM operation. This enhancement also improves the error logging.

### Snapshot-based physical backups are generally available

With [snapshot-based physical backups](../features/snapshots.md) now generally available, you can use this functionality in production environments and enjoy all the benefits of faster restores from snapshots with almost immediate access to data. 

### Dropped support of MongoDB 4.4

Percona Backup for MongoDB drops support of MongoDB 4.4. Existing functionality in Percona Backup for MongoDB remains compatible with MongoDB 4.4 and Percona Server for MongoDB 4.4; however, further enhancements and bug fixes are no longer tested against this version.

### Packaging Changes

* Percona Backup for MongoDB is available on Ubuntu 24.04 (Noble Numbat).
* Percona Backup for MongoDB is no longer supported for Ubuntu 18.04 (Bionic Beaver), Debian 10 and Red Hat Enterprise 7 and derivatives as these operating systems reached End-Of-Life.

> 

Reviewer: ## Join Percona Squad!
Participate in monthly SWAG raffles, get early access to new product features, and invite-only ”ask me anything” sessions with database performance experts. Interested? Fill in the form at [squad.percona.com/mongodb](http://squad.percona.com/mongodb)

## Changelog

### New Features

> * [PBM-1228](https://perconadev.atlassian.net/browse/PBM-1228) - Support multiple storage for backups and restores.

Reviewer 2: Why https://perconadev.atlassian.net/browse/PBM-1300 didn't make it to new features?

Author: Do we add Epics? Or should we? @radoslawszulgo what do you say, shall we add Epics and implementation tickets to RN or go with either ticket type?

Reviewer: IMHO it's not needed. We've already described this in the highlights. We don't need to mention it in release notes unless we provide valuable information in the Jira ticket.

* [PBM-1332](https://perconadev.atlassian.net/browse/PBM-1332) - Fixed the issue with physical backup not being completely removed from the storage after executing `pbm delete-backup` or `pbm cleanup` commands.

* [PBM-1341](https://perconadev.atlassian.net/browse/PBM-1341) - Fixed the issue with failing physical restore on a node shutdown stage by retrying the shutdown operation.

* [PBM-1343](https://perconadev.atlassian.net/browse/PBM-1343) - Fixed the issue with `pbm-agent` not being able to start oplog slicing by releasing the lock before exiting the function to allow pbm-agent on another node to continue the slicing. (Thanks to Tomasz Spyrka for reporting the issue and contribution)

* [PBM-1344](https://perconadev.atlassian.net/browse/PBM-1344) - Continue PITR oplog slicing from a fresh full backup if PBM cannot catch up with old oplog slices on the storage.

* [PBM-1350](https://perconadev.atlassian.net/browse/PBM-1350) - Hide password from log output

* [PBM-1360](https://perconadev.atlassian.net/browse/PBM-1360) - Drop the `pbm.Lock` collection during physical restore to prevent restoring excessive locks 

> * [PBM-1364](https://perconadev.atlassian.net/browse/PBM-1364) - Ensure PITR routine is fully operational after the cluster leader restart

Reviewer 2: ```suggestion
```
Internal ticket for epic, no need to put in RN

* [PBM-1344](https://perconadev.atlassian.net/browse/PBM-1344) - Continue PITR oplog slicing from a fresh full backup if PBM cannot catch up with old oplog slices on the storage.

* [PBM-1350](https://perconadev.atlassian.net/browse/PBM-1350) - Hide password from log output

* [PBM-1360](https://perconadev.atlassian.net/browse/PBM-1360) - Drop the `pbm.Lock` collection during physical restore to prevent restoring excessive locks 

* [PBM-1364](https://perconadev.atlassian.net/browse/PBM-1364) - Ensure PITR routine is fully operational after the cluster leader restart

* [PBM-1367](https://perconadev.atlassian.net/browse/PBM-1367) - Disallow making backups of deployments with sharded time series collections

* [PBM-1239](https://perconadev.atlassian.net/browse/PBM-1239) - Fixed the issue with logical backups failing on GCS by using the libraries that support retries

* [PBM-1348](https://perconadev.atlassian.net/browse/PBM-1348) - Fixed the incorrect behavior of the shell when using the `--wait` for the JSON output in pbm commands

> * [PBM-1377](https://perconadev.atlassian.net/browse/PBM-1377) - Fixed the  issue with backups failing to start when there is insufficient oplog range for PITR

Reviewer 2: ```suggestion
* [PBM-1377](https://perconadev.atlassian.net/browse/PBM-1377) - Fixed the issue with backups failing to start by implementing measures to avoid conflicts with the PITR procedure
```

This release provides the following features and improvements:

### Multiple storages for backups and restores

You can now define multiple storages for making backups and instruct PBM on what storage to save a backup: on the main or on any additional (external) one. This ability helps you save on data transfer costs when using cloud storage, as well as enables you to follow closely with the requirements of your organization’s backup policy.

To learn more, read [Multiple storages for backups](../features/multi-storage.md
).

### Adjust node priority for point-in-time recovery oplog slicing

You can now redefine node priorities not only for making backups but also for point-in-time recovery oplog slices. For example, you can instruct PBM to save oplog slices either from a dedicated node or from nodes in the geographically closest location, thus reducing network latency. 

See the [Adjust node priority for oplog slices](../usage/pitr-tutorial.md#adjust-node-priority-for-oplog-slices) section for guidelines.

### Additional control over PBM command execution

You can now configure the waiting time for a command execution by passing the `--wait-time` flag for the following commands:

> * pbm config --force-resync --wait --wait-time="5m"

Reviewer 3: Please consider to add the new `--wait-time` for all commands where it is supported under the `reference` page. Each of the commands there have a table with the available flags: `--wait-time` will need to be added there, so people who read the command reference know it exists. Add, as usual, a sentence like "Available starting with version 2.6.0.". Thanks!

Author: Addressed in https://github.com/percona/pbm-docs/pull/192

Reviewer 3: Thanks!

* pbm backup --wait --wait-time="5m"
* pbm restore --wait --wait-time="5m"
* pbm oplog-replay --wait --wait-time="5m"
* pbm profile add --sync --wait --wait-time="5m"
* pbm profile remove --wait --wait-time="5m"
* pbm profile sync --wait --wait-time="5m"
* pbm delete-pitr --older-than --wait --wait-time="5m"
* pbm cleanup --older-than --wait --wait-time="5m"

This way you have more control over the PBM operation. This enhancement also improves the error logging.

### Snapshot-based physical backups are generally available

> With [snapshot-based physical backups](../features/snapshots.md) now generally available, you can use this functionality in production environments and enjoy all the benefits of faster restores from snapshots with almost immediate access to data. 

Reviewer 3: Hi! In `snapshots.md` page, under `Considerations`, point 1. Please consider to mention that the feature is GA starting from v2.6.0. Thanks!

Author: Hi! Addressed in https://github.com/percona/pbm-docs/pull/199/files by removing the tech preview label

Reviewer 3: Thanks!

* [PBM-1367](https://perconadev.atlassian.net/browse/PBM-1367) - Disallow making backups of deployments with sharded time series collections

> * [PBM-1239](https://perconadev.atlassian.net/browse/PBM-1239) - Fixed the issue with logical backups failing on GCS by using the libraries that support retries

Reviewer 2: ```suggestion
* [PBM-1239](https://perconadev.atlassian.net/browse/PBM-1239) - Fixed the issue with logical backups failing on GCS by utilizing libraries that provide retry functionality.
```

> * [PBM-1348](https://perconadev.atlassian.net/browse/PBM-1348) - Fixed the incorrect behavior of the shell when using the `--wait` for the JSON output in pbm commands

Reviewer 2: ```suggestion
* [PBM-1348](https://perconadev.atlassian.net/browse/PBM-1348) - Fixed the incorrect behavior of PBM CLI where the `--wait` option was causing incorrect behavior with `pbm backup` command while generating JSON output
```
